In [1]:
import numpy as np
import pandas as pd
import yfinance as yf
from scipy.stats import norm
import matplotlib.pyplot as plt
from IPython.display import display, HTML

np.random.seed(42)

In [35]:
pd.set_option('display.float_format', '{:.2%}'.format)

tickers = ['AAPL', 'F', 'AMZN', 'GOOGL', 'NFLX', 'META', 'DIS', 'AMD', 'NVDA', 'TSLA', 'SHOP', 'GM', 'PYPL', 'ROKU', 'ADBE', 'MNST', 'T', 'INTC', 'AXP']
start_date = '2024-01-01'
end_date = '2025-04-24'

data = yf.download(tickers, start=start_date, end=end_date, progress=False)['Close'][tickers]
rt = data.pct_change().dropna()

mu = rt.mean() * 252
sigma = rt.std() * np.sqrt(252)

metricas = pd.DataFrame({
    'Rendimiento': mu,
    'Volatilidad': sigma,
})
metricas.T

Ticker        AAPL      F   AMZN  GOOGL   NFLX   META    DIS     AMD   NVDA  \
Rendimiento 12.49% -2.00% 19.45% 14.10% 67.69% 39.39%  2.06% -18.99% 75.47%   
Volatilidad 30.33% 37.21% 32.03% 30.74% 33.22% 39.62% 29.53%  53.19% 58.53%   

Ticker        TSLA   SHOP     GM   PYPL   ROKU    ADBE   MNST      T    INTC  \
Rendimiento 23.93% 32.94% 25.54%  8.45% -8.41% -31.42%  3.59% 43.96% -46.47%   
Volatilidad 68.95% 58.21% 34.75% 37.81% 60.45%  37.16% 24.13% 22.10%  58.38%   

Ticker         AXP  
Rendimiento 30.70%  
Volatilidad 29.91%

Si usamos un umbral de 35% mínimo de volatilidad las potenciales serían
- F
- META
- NVDA
- TSLA
- SHOP
- GM
- PYPL
- NFLX
- AXP
- AAPL